From ad1d7a89a8c51a00afd4b77820123d73f3a340c4 Mon Sep 17 00:00:00 2001
From: Kalle Lampila <kallex.lampila@intel.com>
Date: Tue, 30 Apr 2013 15:27:19 +0300
Subject: Camera service: fix deadlock

BZ: 104720

Release lock before return from CameraService::dump().

Category: aosp improvement
Domain: camera
Origin: internal
Upstream-Candidate: yes

Change-Id: Ia20b0921fc675a284f3b7f8d26e0e3596b5a8774
Orig-Change-Id: I89d75f13bf82490929eafa1da9e88e7023abfb49
Signed-off-by: Kalle Lampila <kallex.lampila@intel.com>
---
 services/camera/libcameraservice/CameraService.cpp |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/services/camera/libcameraservice/CameraService.cpp b/services/camera/libcameraservice/CameraService.cpp
index 27e48d7..274fe28 100644
--- a/services/camera/libcameraservice/CameraService.cpp
+++ b/services/camera/libcameraservice/CameraService.cpp
@@ -1209,6 +1209,7 @@ status_t CameraService::dump(int fd, const Vector<String16>& args) {
         if (!mModule) {
             result = String8::format("No camera module available!\n");
             write(fd, result.string(), result.size());
+            if (locked) mServiceLock.unlock();
             return NO_ERROR;
         }
 
-- 
1.7.4.1

